### DISCUSSION TITLE
The space-x-4 class doesn't working correctly with the flex-wrap class

### DISCUSSION THREAD

**Discussion by buismaarten** [tailwindlabs/tailwindcss#1735] [Help] [answered]
I want to try out the new `space-x-4` class to prevent using the margin or padding technique. I also want to use the `flex-wrap` class to prevent columns going full width.

Sadly the result is not as I expected.

Example: https://codepen.io/buismaarten/pen/PoPRqdJ

**Comment by buismaarten** [comment]
Solved this problem by myself using:
- https://tailwindcss.com/docs/grid-template-columns
- https://tailwindcss.com/docs/gap

**Comment by frederikhors** [comment]
@buismaarten same problem here.

Is there a way to fix it without css-grid? Using flexbox?

**Comment by frederikhors** [comment]
Can you please remove the picked answer from this? Is not a proper "solution" to the problem.

**Comment by frederikhors** [comment]
No answer yet. This is a problem. I'll open an issue.

**Comment by frederikhors** [comment]
https://github.com/tailwindcss/tailwindcss/issues/1975

**Comment by hacknug** [comment, marked as answer]
Just use Ben's plugin or use paddings with negative margins: https://github.com/benface/tailwindcss-gap

**Reply by frederikhors** [reply]
@hacknug thanks! Here it is explained why using negative margins is wrong: https://github.com/tailwindcss/tailwindcss/pull/1584#issuecomment-619381664

**Reply by hacknug** [reply]
Nah, nothing wrong with using CSS. Every approach has its drawbacks and its your job to understand how they all work.

**Reply by adamwathan** [reply]
Yep it's totally a fine solution (and the right solution when things need to wrap), just adds too much complexity for the 90% use case to be the default implementation IMO.

**Reply by Silberlachs** [reply]
a plugin? THIS is your solution?

**Reply by hacknug** [reply]
> a plugin? THIS is your solution?

You can now use `gap` for this if you don't need to support old versions of browsers. Nothing wrong with using a plugin, that's what the plugin API is for… 🤷‍♂️

**Comment by Youniteus** [comment]
I ran into the same problem.

I was thinking about writing a plugin that adds classes like:
```
.flex-col-gap-4 {
  margin-left: -1rem;
  margin-right: -1rem;
  > * {
    padding-left: 1rem;
    padding-right: 1rem;
  }
}
```

**Comment by aeadedoyin** [comment]
What I do most times, is to use padding for the direct child div of the flexed(wrapped) parent, then have an inner div handle the content of each col.

```
div.flex.flex-wrap>
  div.md:col-6.p-4>
    div.bg-blue-500>  - content here
  div.md:col-6.p-4>
    div.bg-gray-500>  - content here
```

For the **col-*** here's the plugin I use:
https://github.com/karolis-sh/tailwind-bootstrap-grid

**Reply by Silberlachs** [reply]
i ended up using an invisible character alt+255
now i'm waiting until my markup blows up
